src/exercises/pointer-revocation/temporal-control.c

```diff
@@ -30,7 +30,10 @@ fn2(uintptr_t arg)
 
 struct obj {
 	char buf[32];
-	/* Volatile so that the compiler will always reload it */
+	/*
+	 * The following are volatile so that the compiler will always
+	 * reload them.
+	 */
```

Comment:
Hm, that's not really why we make `arg` volatile, I'm assuming it's more to ensure it's persisted to memory in the first place? I also don't want to give the idea that it needs to be volatile to be revoked.

Reply:
Hmm, how about
```
	 * The following are volatile so that the compiler will always
	 * store and reload them rather than optimizing memory access
	 * away.
```

Reply:
So, looking at the compiled output, it seems that for RISC-V, the Dead Store Elimination pass optimises out the store into `arg` but leaves the loads in, so they'll see the zeroes from `calloc`. For CHERI-RISC-V it doesn't think the store is dead, which is surprising. Perhaps it doesn't think `void @free(i8 addrspace(200)*)` is the real `free`.

How about something like:
```C
    /*
     * The following are marked volatile to ensure the compiler doesn't
     * constant propagate fn (making aliasing not work) and to ensure neither
     * stores to them are optimised away entirely as dead due to calling free.
     */
```
which is hopefully less vague in a way that makes it clear that it's not a limit of revocation.